dbt_dry_run/node_runner/incremental_runner.py

```diff
@@ -191,15 +223,21 @@ def run(self, node: Node) -> DryRunResult:
 
         full_refresh = self._get_full_refresh_config(node)
 
-        if result.status == DryRunStatus.SUCCESS and not full_refresh:
-            target_table = self._sql_runner.get_node_schema(node)
-            if target_table:
-                on_schema_change = node.config.on_schema_change or OnSchemaChange.IGNORE
-                handler = ON_SCHEMA_CHANGE_TABLE_HANDLER[on_schema_change]
-                result = self._verify_merge_type_compatibility(
-                    node, sql_with_literals, result, target_table
-                )
-                if result.status == DryRunStatus.SUCCESS:
-                    result = handler(result, target_table)
+        if result.status == DryRunStatus.SUCCESS:
+            if self._is_time_ingestion_partitioned(node):
+                result = self._replace_partition_with_time_ingestion_column(result)
+
+            if not full_refresh:
+                target_table = self._sql_runner.get_node_schema(node)
```

Comment:
If the incremental already exists does dbt behave differently? I don't think dbt can modify the partition keys for tables once they have been created without a full refresh

Reply:
I reverted this line back to the original.